polygeist/tools/cgeist/driver.cc

```diff
@@ -984,11 +994,15 @@ int main(int argc, char **argv) {
 
   // For now, we will work on the device code if SYCL kernels are found and on
   // the host code otherwise.
-  //
+  if (hasDeviceFuncs(deviceModule)) {
+    eraseHostCode(*module);
+    module.get()->setAttr(mlir::gpu::GPUDialect::getContainerModuleAttrName(),
+                          Builder.getUnitAttr());
+  } else {
+    deviceModule.erase();
+  }
+
   // Lower the MLIR to LLVM IR, compile the generated LLVM IR.
-  return hasDeviceFuncs(*deviceModule)
-             ? compileModule<mlir::gpu::GPUModuleOp>(
-                   deviceModule, context, DL, triple, LinkageArgs, argv[0])
-             : compileModule<mlir::ModuleOp>(module, context, DL, triple,
-                                             LinkageArgs, argv[0]);
+  return compileModule<mlir::ModuleOp>(module, context, DL, triple, LinkageArgs,
```

Comment:
Do we now always call `compileModule` with template argument `mlir::ModuleOp`? If so, it no longer need to be a templated function.

Reply:
Yes, I was planning on refactoring in a different PR to push this one, as it fixes a bug and I wanted to make it easier to review (I was planning on not passing the `deviceModule` to the AST consumer as well, but I'd rather do that in a different PR).